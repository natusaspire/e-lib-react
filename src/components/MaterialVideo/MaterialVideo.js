import React, { memo } from 'react';

const MaterialVideo = ({ material }) => (
  <div className="card">
    <div className="card__header">{material.title}</div>
    <div className="content">
      <div className="card__player">
        <video
          controls
          src={material.url}
        >Sorry, your browser does not support embedded video, you can <b><a className="card__link" href={material.url}>download</a></b> it.</video>
      </div>
    </div>
  </div>
);

export default memo(MaterialVideo);
