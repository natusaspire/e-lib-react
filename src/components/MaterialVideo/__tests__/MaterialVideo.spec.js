import React from 'react';
import { shallow } from 'enzyme';

import MaterialVideo from '@/components/MaterialVideo';

describe('<MaterialVideo />', () => {
  it('should render correctly', () => {
    const material = {
      id: 1,
      type: 'video',
      title: 'Material title',
      url: 'video.mp4'
    };

    const wrapper = shallow(<MaterialVideo material={material} />);

    expect(wrapper).toMatchSnapshot();
  });
});
