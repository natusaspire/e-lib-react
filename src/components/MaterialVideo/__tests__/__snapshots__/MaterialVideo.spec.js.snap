// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MaterialVideo /> should render video material 1`] = `
<div
  className="card"
>
  <div
    className="card__header"
  >
    Material title
  </div>
  <div
    className="content"
  >
    <div
      className="card__player"
    >
      <video
        controls={true}
        src="video.mp4"
      >
        Sorry, your browser does not support embedded video, you can
         
        <b>
          <a
            className="card__link"
            href="video.mp4"
          >
            download
          </a>
        </b>
         
        it.
      </video>
    </div>
  </div>
</div>
`;
